### PR TITLE
You can no longer TK throw vending machines

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Back when I made vendors able to squish people when thrown into them (for the purposes of admin buildmode abuse), I ended up making using TK throws on a vendor incredibly powerful. #56063 limited these throws to one tile, but that's still incredibly strong considering you're basically guaranteed to maim whoever you hit with it.

No it's not ided, I observed it being used and even as someone who enjoys seeing my features get used, it was pretty painful to watch.

## Why It's Good For The Game
There's a million other ways to trigger the funny vendor tip meme, we don't need to allow it in such a dumb, unsubtle way.

## Changelog
:cl: Ryll/Shaps
balance: Vending machines can no longer by thrown via TK
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
TWO DUBS

**Review by Mothblocks — COMMENTED** [review]
I remember when this was nerfed, a `tk_range` was added. If that's on all machinery, let me know. If it's on the vending machine specifically, remove it. That's my only point of feedback.

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -1007,6 +1008,9 @@ GLOBAL_LIST_EMPTY(vending_products)
 
 	tilt(L)
 
+/obj/machinery/vending/attack_tk_grab(mob/user)
+	to_chat(user, "<span class='warning'>[src] seems to resist your mental grasp!</span>")
```
Comment (TemporalOroboros) [inline code/modules/vending/_vending.dm:1012]: Could this just be a no-op? Maybe with a comment on it for explanation as to _why_ it's a no-op. Using TK on something and having nothing happen is much more immersive than using TK on something and having an OOC message appear.

Comment (Ryll-Ryll) [reply code/modules/vending/_vending.dm:1012]: Failing silently with no indication of why is much worse than showing a simple (not at all OOC) message saying you can't do it and moving on with your life

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -59,6 +59,7 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	payment_department = ACCOUNT_SRV
 	light_power = 0.5
 	light_range = MINIMUM_USEFUL_LIGHT_RANGE
+	tk_throw_range = 0
```
Comment (Mothblocks) [inline code/modules/vending/_vending.dm:62]: This shouldn't be necessary? I meant remove it if it existed on the vending machine type.

**Comment by Ratvar** [comment]
Out of scope of PR, why keep TK at all if it causes issues like that every single time and is used for grief or uninteractive through wall fights?

**Comment by Ryll-Ryll** [comment]
> Out of scope of PR, why keep TK at all if it causes issues like that every single time and is used for grief or uninteractive through wall fights?

That's a pretty massive leap considering how niche this balance issue is and the fact that it only came about because i was dumb and didn't think things through when i did the initial vendor throw change

**Comment by Hadzabadza** [comment]
> I remember when this was nerfed, a `tk_range` was added. If that's on all machinery, let me know. If it's on the vending machine specifically, remove it. That's my only point of feedback.

It also applies to crates and lockers

**Comment by Mothblocks** [comment]
I meant in that, I know it's on the base of /obj or something, but I wanted to know if vending machines specifically had a special value. It looks like they don't.

**Comment by Mothblocks** [comment]
i almost died to this so i get to merge this on i died clause

**Comment by Hadzabadza** [comment]
> i almost died to this so i get to merge this on i died clause

So now that you can't even lift vendors anymore, can we get proper throwing ranges for harmless crates and lockers back?

**Comment by Mothblocks** [comment]
if that was removed in the same pr then sure why not

**Pull request merged**